packages/ts-transform/src/utils/css-transform.tsx

```diff
@@ -19,33 +20,42 @@ const minify = () => {
     });
 };
 
-const COMMA_CHAR_CODE = 58;
+const isPreviousSelectorCombinatorType = (selector: selectorParser.Node) => {
+  const previousSelector = selector.prev();
 
+  return previousSelector && previousSelector.type === 'combinator';
+};
+const prependNestingTypeToSelector = (selector: selectorParser.Node) => {
+  const { parent } = selector;
+
+  if (parent) {
+    const nesting = selectorParser.nesting();
+
+    parent.insertBefore(selector, nesting);
+  }
+};
+const stringifySelectorParserRoot = (parserRoot: selectorParser.Root) =>
+  parserRoot.reduce<string[]>((memo, selector) => [...memo, String(selector)], []).join(',\n');
 const parentOrphenedPseudos = plugin('parent-orphened-pseudos', () => {
   return (root) => {
     root.walkRules((rule) => {
-      if (rule.selector.includes(':')) {
-        const newSelector = rule.selector
-          .replace(/\s+/g, ' ')
-          .split(', ')
-          .map((part) => {
-            if (part.match(/^. /)) {
-              // If the selector has one characters with a space after it, e.g. "> :first-child" then return early.
-              return part;
-            }
-
-            if (part.charCodeAt(0) === COMMA_CHAR_CODE) {
-              // If the selector starts with a colon prepend an "&"!
-              return part.replace(/^:| :/g, '&:');
-            }
-
-            // Nothing to do - cya!
-            return part;
-          })
-          .join(',\n');
-
-        rule.selector = newSelector;
+      const { selector: ruleSelector } = rule;
+
+      if (!ruleSelector.includes(':')) {
+        return;
       }
+
+      const selectorParserRoot = selectorParser((selectors) => {
+        selectors.walkPseudos((selector) => {
+          if (isPreviousSelectorCombinatorType(selector)) {
```

Comment:
hmm did we not need to check if the selector is a pseudo anymore? 🤔

Reply:
`selectors .walkPseudos`?

Reply:
OH

cool!!!